deployability/modules/allocation/generic/provider.py

```diff
@@ -154,26 +154,39 @@ def _generate_instance_id(prefix: str) -> str:
         return f"{prefix}-{uuid.uuid4()}".upper()
 
     @classmethod
-    def _get_os_specs(cls) -> dict:
+    def _get_os_specs(cls, composite_name: str) -> dict:
         """
         Gets the OS specifications for the provider.
+        composite_name (str): The name of the composite OS.
 
         Returns:
             dict: A dictionary containing the OS specifications for the provider.
         """
         with open(cls.OS_PATH, "r") as f:
-            return yaml.safe_load(f).get(cls.provider_name)
+            version_available = []
+            os_list = yaml.safe_load(f).get(cls.provider_name)
+            for os in os_list:
+                if os.split("-")[1] == composite_name.split("-")[1]:
+                    version_available.append(os)
+                if str(os) == composite_name:
+                    return os_list[os]
+
+            raise ValueError(f"OS {composite_name} not available for provider {cls.provider_name}. Available versions are {version_available}")
 
     @classmethod
-    def _get_size_specs(cls) -> dict:
+    def _get_size_specs(cls, size: str) -> dict:
         """
         Gets the size specifications for the provider.
+        size (str): The name of the size.
 
         Returns:
             dict: A dictionary containing the size specifications for the provider.
         """
         with open(cls.SIZE_PATH, "r") as f:
-            return yaml.safe_load(f).get(cls.provider_name)
+            size_list = yaml.safe_load(f).get(cls.provider_name)
+            for size in size_list:
+                if size == size:
+                    return size_list[size]
```

Comment:
We could incorporate a raise if the size is not found, just as it is in _get_os_specs

Reply:
This was validated in models.py
```
class InputPayload(BaseModel):
    size: Literal['micro', 'small', 'medium', 'large', None] = None
```